src/mudlet.cpp

```diff
@@ -2378,19 +2369,20 @@ QString mudlet::readProfileData( const QString& profile, const QString& item )
 // this slot is called via a timer in the constructor of mudlet::mudlet()
 void mudlet::startAutoLogin()
 {
-    QStringList hostList = QDir(QDir::homePath()+"/.config/mudlet/profiles").entryList(QDir::Dirs, QDir::Name);
-    hostList.removeAll(".");
-    hostList.removeAll("..");
-    for( int i = 0; i< hostList.size(); i++ )
-    {
-        QString item = "autologin";
-        QString val = readProfileData( hostList[i], item );
-        if( val.toInt() == Qt::Checked )
-        {
-            doAutoLogin( hostList[i] );
+    QStringList hostList = QDir(QDir::homePath() + "/.config/mudlet/profiles").entryList(QDir::Dirs | QDir::NoDotAndDotDot, QDir::Name);
+    bool openedProfile = false;
+
+    for (auto host : hostList) {
+        QString val = readProfileData(host, QStringLiteral("autologin"));
+        if (val.toInt() == Qt::Checked) {
```

Comment:
If the value has __not ever__ been set, `val` will be empty - so `val.toInt(bool * isOk)` will fail (The boolean `isOk` points at, if non-null, would be set to `false` instead of `true`) and it _just so happens_ that the `enum QtCheckState` has the value of `0`for `Qt::Unchecked` but technically (bloody unlikely, I know 🙄 ) Qt _could_ change the enumerated values in the future and then `0` might be the `Qt::Checked` value instead - and then every profile would get fired up...!

I wonder if __Coverity__ would pick up on this?

The reason I'm going on about it is that I vaguely recall having an issue with, a `qsort` `lessThan` functor which was doing a comparison on two strings that both _might_ be numbers - the behaviour if both of them were strings was broken because if they were not numbers the `QString::toInt()` was quietly returning a zero for both of them which WERE being treated as numbers when a string comparison should have been done instead... and __that__ bug was caused by failing to account for `toInt()` not succeeding either!

tl;dr; Basically I am not sure it is wise to depend on the consistency of __numeric__ values of `enums` which have values expressed as `named constants` when the allocation of _values_ to elements of the _enum_ is not within our perview...